common/js/rechat.js

```diff
@@ -384,13 +394,55 @@ ReChat.Playback.prototype._formatChatMessage = function(messageData) {
   } else {
     colon.text(':');
   }
+
   from.text(messageData.from.replace('\\s', ' '));
   var messageHtml = this._replaceEmoticonsByRanges(messageText, messageData.emotes);
   message.html(messageHtml);
-  line.append(from).append(colon).append(' ').append(message);
+  line.append(badges).append(from).append(colon).append(' ').append(message);
   return line;
 };
 
+ReChat.Playback.prototype._applyMessageBadges = function(messageData, badges) {
+  if (messageData.from.toLowerCase() == this._channelName) {
+    var badgeContent = this._buildBadge().addClass('broadcaster').prop('title', 'Broadcaster');
+    badges.append(badgeContent).append(' ');
+  }
+  if (messageData.usertype != null) {
+    var badgeContent = this._buildBadge();
+    switch (messageData.usertype) {
+      case "mod":
+        badgeContent.addClass('moderator').prop('title', 'Moderator');
+        break;
+      case "global_mod":
+        badgeContent.addClass('global-moderator').prop('title', 'Global Moderator');
+        break;
+      case "admin":
+        badgeContent.addClass('admin').prop('title', 'Twitch Admin');
+        break;
+      case "staff":
+        badgeContent.addClass('staff').prop('title', 'Twitch Staff');
+        break;
+    }
+    badges.append(badgeContent).append(' ');
+  }
+  if (messageData.turbo) {
+    var badgeContent = this._buildBadge().addClass('turbo').prop('title', 'Twitch Turbo');
+    badges.append(badgeContent).append(' ');
+  }
+  if (messageData.subscriber && this._subscriberUrl != null) {
+    var badgeContent = this._buildBadge().addClass('subscriber')
+                                         .prop('title', 'Subscriber')
+                                         .css('background-image', 'url(' + this._subscriberUrl + ')');
+    badges.append(badgeContent).append(' ');
+  }
+}
+
+ReChat.Playback.prototype._buildBadge = function() {
+  return $('<div>').addClass('float-left badge ').attr({
+    style: 'margin: 1px 3px 0 0; float: left!important;' // BTTV style overriding
```

Comment:
Here is the same thing about `!important`. I also included the `margin` there because I didn't want to do an `$.attr()` call for the `!important` and a `$.css()` call for the `margin`.